shuup/campaigns/models/campaigns.py

```diff
@@ -417,11 +426,6 @@ def has_been_used(self, usage_count=1):
         """ See if code is used the times given """
         return CouponUsage.objects.filter(coupon=self).count() >= usage_count
 
-    def save(self, **kwargs):
-        if Coupon.objects.filter(code__iexact=self.code, active=True).exclude(pk=self.pk).exists():
```

Comment:
You should check here the same logic when saving the code as we merchant can, for some weird reason, change the coupon code OR, just we just disable the coupon code change, once created, do not change anymore.

Also, in the multi-vendor system, several shops sharing the same code and allowing all shops changing the code is some what dangerous. I mean, let's suppose I have a shop and create a coupon code and you also have a shop. You have bad intentions and simply change my coupon options, like usage limit or even the active property. Does that seem right? Maybe we should also have a coupon x shop relation here.